lib/miq_automation_engine/service_models/miq_ae_service_container_deployment.rb

```diff
@@ -0,0 +1,192 @@
+      unless subscribed.include? "This system is already registered"
+        pool_id = perform_agent_commands(deployment_master, user, ["sudo subscription-manager list --available --matches='#{rhsm_sku}' --pool-only"])[:stdout].split("\n").first.delete("\r")
+        perform_agent_commands(deployment_master, user, ["sudo subscription-manager attach --pool='#{pool_id}'",
+                                                         'sudo subscription-manager repos --disable="*"',
+                                                         'sudo subscription-manager repos --enable="rhel-7-server-rh-common-rpms" --enable="rhel-7-server-rpms" --enable="rhel-7-server-extras-rpms" --enable="rhel-7-server-ose-3.2-rpms"'
+                                                ])
+      end
+      if nodes_subscription_needed?
+        perform_scp(deployment_master, user, "rhel_subscribe_inventory.yaml", "rhel_subscribe_inventory.yaml")
+      end
+    end
+
+    def subscribe_cluster(deployment_master, user)
+      perform_agent_commands(deployment_master, user, ["sudo mv ~/rhel_subscribe_inventory.yaml /usr/share/ansible/openshift-ansible/"])
+      run_playbook_command(deployment_master, user, "ansible-playbook /usr/share/ansible/openshift-ansible/playbooks/byo/rhel_subscribe.yml -i "\
+                         "/usr/share/ansible/openshift-ansible/rhel_subscribe_inventory.yaml 1> /tmp/ansible.log 2> /tmp/openshift-ansible.log.2 < /dev/null")
+    end
+
+    def nodes_subscription_needed?
+      container_deployment_nodes.count > 1
+    end
+
+    def ssh_user
+      ssh_auth.userid
+    end
+
+    def add_deployment_provider(options)
+      object_send(:add_deployment_provider, options)
+    end
+
+    def regenerate_ansible_inventory
+      object_send(:generate_ansible_yaml)
+    end
+
+    def regenerate_ansible_subscription_inventory
+      object_send(:generate_ansible_inventory_for_subscription)
+    end
+
+    def add_automation_task(task)
+      ar_method do
+        @object.automation_task = AutomationTask.find_by_id(task.id)
+        @object.save!
+      end
+    end
+
+    def customize(options = {})
+      ar_method do
+        if @object.customizations.empty?
+          @object.customizations = {:agent => {}}
+        end
+        @object.customizations[:agent].merge!(options)
+        @object.save!
+      end
+    end
+
+    def perform_scp(ip, username, local_path, remote_path)
+      Net::SCP.upload!(ip,
+                       username,
+                       local_path,
+                       remote_path,
+                       :ssh => {:key_data => ssh_auth.auth_key})
+    end
+
+    def perform_agent_commands(ip, username, commands = [])
+      ssh = LinuxAdmin::SSH.new(ip, username, ssh_auth.auth_key)
+      ssh.perform_commands(commands)
+    end
+
+    def check_connection(deployment_host_ip, username, nodes_ips)
+      agent = LinuxAdmin::SSHAgent.new(ssh_auth.auth_key, "/tmp/ssh_manageiq/ssh_manageiq_#{id}")
```

Comment:
This agent is the local one it is only used for the first ssh connection checks, all the rest of the commands don't use the agent beside the playbooks that uses the remote agent.